src/resources/cpp/language-basics.md

```diff
@@ -1,15 +1,307 @@
+representation. The standard does not specify if `char` is signed or unsigned, but its signedness is determined by the
+compiler and is usually the most optimal representation for a single byte character on the platform. On x86-64, char is
+typically signed, whereas on ARM64, it is typically unsigned.
+
+For characters that are not able to be represented in a single byte, C++ provides other distinct types for wider
+character storage.
+
+- `wchar_t` - A character type that is typically wider than the `char` type. On \*nix platforms, it is typically 32 bits
+  in width. On windows platforms, it is typically 16 bits in width.
+- `char8_t` - A character type that can represent any UTF-8 character representation.
+- `char16_t` - A character type that can represent any UTF-16 code point.
+- `char32_t` - A character type that can represent any UTF-32 code point.
+
+### Floating Point Types
+
+A floating point type is a real number that can have a decimal part. However, floating point types cannot represent all
+decimal numbers, as computers are only capable of processing numbers of a fixed size. Thus, computers use various
+representations to approximate decimals.
+
+There are three fundamental floating point types in C++, `float`, `double`, and `long double`. The range of
+representable values for a `double` is required to be at least as large as that of a `float`, and the `long double` must
+have a range at least that of a `double`. The language does not provide many strong guarantees about the properties of
+the various floating point types.
+
+::: info IEEE-754 Floating Point Model
+
+Because C++ needs to support many different types of computers, it does not enforce that floating points are represented
+in a specific way. For the sake of this tutorial, we will assume that you are working on a device that uses the IEEE-754
+floating point specification. This is the case for almost all consumer hardware. The rest of the tutorial will make the
+assumption that a `float` is an IEEE-754 binary32 format float and that a `double` is an IEEE-754 binary64 format float.
+
+| Type     | Range                  | Width |
+| -------- | ---------------------- | ----- |
+| `float`  | $\pm 3.402 * 10^{38}$  | 32    |
+| `double` | $\pm 1.797 * 10^{308}$ | 64    |
+
+The type `long double` has the restriction that its range must be at least as large as a `double`. On many x86-64
+platforms, it is implemented as an IEEE-754 binary64 extended format, where the width of the type is 80 bits. However,
+MSVC is an exception to this rule, where it is only 64 bits. On other platforms, it may be implemented as a IEEE-754
+binary128 format float.
+
+:::
+
+### Literals
+
+Literals are a way that C++ allows us to provide a value directly in the code. You may recall above where the code
+snippets were creating variables using numbers in the code. Those numbers are referred to as literals.
```

Comment:
This should probably link to that section via anchor link, or it should be closer to the section that talks about variables and stuff, so you don't need a long-distance backwards reference.